api/qan/collector.proto

```diff
@@ -137,9 +170,15 @@ message MetricsBucket {
   float m_tmp_table_sizes_min = 110;
   float m_tmp_table_sizes_max = 111;
   float m_tmp_table_sizes_p99 = 112;
+  //
   // boolean metrics
+  //
+  // FIXME(askomorokhov) why they are float then?
```

Comment:
@askomorokhov ^

Reply:
This is boolean metrics in MySQL Percona Server slow log.
Values passes in gRCP will be int.
We desided to store all metrics in ClickHouse as float32.
What is your opinion? Does we win more in speed or memory with additional converting  int<->float.?

Reply:
`float32(1)` is like 100x faster than float decoding in protobuf.

Reply:
But that metrics are actually boolean? Count? Sum? That's unlikely :)

Reply:
> float32(1) is like 100x faster than float decoding in protobuf. 

sorry, did not get this your comment.
For me at a glance they seems quite similar in speed:
https://developers.google.com/protocol-buffers/docs/encoding
https://github.com/golang/protobuf/blob/master/proto/decode.go

> But that metrics are actually boolean? Count? Sum? That's unlikely :)

Let's  imaging that in slow log we have 10 entry of same query.
So, num_query = 10. if query use Query cache that QC_Hit will be present in slowlog entry.
```
# QC_Hit: No  Full_scan: Yes  Full_join: No  Tmp_table: No  Tmp_table_on_disk: No
# Filesort: No  Filesort_on_disk: No  Merge_passes: 0
```
or 
```
# QC_Hit: Yes  Full_scan: Yes  Full_join: No  Tmp_table: No  Tmp_table_on_disk: No
# Filesort: No  Filesort_on_disk: No  Merge_passes: 0
```
Hence m_qc_hit_cnt is counter how many time QC_Hit "flag" appear in given segment of slowlog. Zero is NULL here.
And m_qc_hit_sum is counter how many time QC_Hit was equal to "Yes".